docs/topics/scheduler.rst

```diff
@@ -0,0 +1,59 @@
+.. _topics-scheduler:
+
+=========
+Scheduler
+=========
+
+The Scheduler receives requests from the :ref:`Engine <component-engine>`
+and enqueues them into disk and/or disk queues
+(:meth:`scrapy.core.scheduler.Scheduler.enqueue_request`).
+It also gets those requests and feeds them back when the engine requests
+them (:meth:`scrapy.core.scheduler.Scheduler.next_request`).
```

Comment:
I think it may be better to make some changes to this introduction:

- I think we should make it clear what the ultimate sources of requests are, possibly in an additional paragraph after this introduction: `start_requests` or `start_urls`, callbacks, methods A and B of spider middlwares, methods C and D of downloader middlewares.
- I think how the scheduler stores requests should be described in a later paragraph. And we need to be careful, because many of it seems specific to the default scheduler, a custom scheduler could handle request storage in a completely different way. Maybe here we should simply say that it is responsible for storing those request, without going into the specifics of how the default scheduler stores them. And we could then add a section about the default scheduler, and cover how it handles storage there.
- I think we should emphasize that the scheduler determines the order in which those requests are then fed into the engine.